### PR TITLE
Add LinearBarSeries in OxyPlot and OxyPlot.Wpf

### LINKED ISSUE #506: Add a series that can display bars with a LinearAxis or a DateTimeAxis
I implemented a BarDataPointSeries class to display bars with a LinearAxis or a DateTimeAxis.
I can create a pull request to add this feature to OxyPlot.

I only implemented the WPF version but I can support more frameworks if needed.
It goes without saying that the series class name can be changed.

### PR DESCRIPTION
**Pull request by ocoanet**
Resolve #506

### CODE REVIEW

File: Source/OxyPlot/OxyPlot.csproj [hunk 1]
```diff
@@ -63,6 +63,7 @@
     <Compile Include="PlotModel\IPlotElement.cs" />
     <Compile Include="PlotModel\IPlotModel.cs" />
     <Compile Include="Rendering\OxySizeExtensions.cs" />
+    <Compile Include="Series\BarSeries\LinearBarSeries.cs" />
```
Comment (objorke) [inline Source/OxyPlot/OxyPlot.csproj]: the other project files in this directory should also be updated/kept in sync with this

**Comment by objorke** [comment]
The change log should also be updated! :-)

**Comment by objorke** [comment]
Please add an example in the ExampleLibrary - then we can test this series on all platforms!

**Comment by ocoanet** [comment]
Done, done and done.

**Comment by ocoanet** [comment]
The AppVeyor build failure does not seem to be related to my commits. Tell me if I need to fix something.

**Comment by ocoanet** [comment]
I re-merged develop into the branch to get a passing build. Do you need anything else?

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ All notable changes to this project will be documented in this file.
 - Added WPF Wrapper for BoxPlot (#434)
 - Added capability to display mean value to BoxPlot (#440)
 - Added TitleToolTip to PlotModel (#508)
+- Added LinearBarSeries for WPF (#507)
```
Comment (objorke) [inline CHANGELOG.md]: should not be for WPF only

File: Source/OxyPlot/Series/BarSeries/LinearBarSeries.cs [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+        /// <summary>
+        /// Gets the nearest point.
+        /// </summary>
+        /// <param name="point">The point.</param>
+        /// <param name="interpolate">interpolate if set to <c>true</c> .</param>
+        /// <returns>A TrackerHitResult for the current hit.</returns>
+        public override TrackerHitResult GetNearestPoint(ScreenPoint point, bool interpolate)
+        {
+            var rectangleIndex = this.rectangles.BinarySearch(0, this.rectangles.Count, new OxyRect(), new NearestPointComparer(point));
```
Comment (objorke) [inline Source/OxyPlot/Series/BarSeries/LinearBarSeries.cs]: is there a binary search extension method that takes a Func? I think that would be cleaner than creating an `IComparer<OxyRect>` class.

**Comment by objorke** [comment]
This looks mostly OK to me! Could you squash the commits into 1, please? https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by ocoanet** [comment]
I cannot squash the commits because they are already pushed. Do you want me to create a new branch with a single commit?

**Comment by objorke** [comment]
I think squashing pushed commits on a fork is possible and OK: http://stackoverflow.com/questions/5667884/how-to-squash-commits-in-git-after-they-have-been-pushed

**Comment by ocoanet** [comment]
Ok, done.

**Comment by objorke** [comment]
thanks for the contribution! Sorry for the merge delay... :-)

**Pull request merged**